docs/data/data-grid/editing/editing.md

```diff
@@ -4,266 +4,408 @@ title: Data Grid - Editing
+{{"demo": "ValidateServerNameGrid.js", "bg": "inline", "defaultCodeOpen": false}}
 
-The demo lets you edit the ratings by double-clicking the cell.
+> ⚠ If an interaction that saves and stops editing (e.g. pressing <kbd class="key">Enter</kbd>) occurs while processing the props, the changes made are discarded and the cell will immediatelly exit the edit mode.
+> To avoid losing the changes is important to communitate users about what is happening.
+> You can use the `isProcessingProps` prop to provide feedback while waiting the server to respond.
 
-{{"demo": "RenderRatingEditCellGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+## Persistence
 
-### Edit using external button [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+After executing one of the [stop editing](#stop-editing) interactions, the `processRowUpdate` prop will be called.
+Use this prop to send the new values to the server and save it into a database or other storage method.
+The prop is called with two arguments: the first being the updated row with the new values, and after passing through the value setter, and the second is the row before the current cell or row has entered the edit mode (a.k.a the original row).
+Once the row was saved, `processRowUpdate` must return the row object that will be used to update the internal state.
+This value returned is used as argument to a call to `apiRef.current.updateRows`.
 
-You can override the default [start editing](#start-editing) triggers using the [`event.defaultMuiPrevented`](/components/data-grid/events#disabling-the-default-behavior) on the synthetic React events.
+If while calling `processRowUpdate` you need to cancel the save process, like when a database validation fails or the user wants to reject the changes, there are two options:
 
-{{"demo": "StartEditButtonGrid.js", "bg": "inline", "disableAd": true}}
+- Reject the promise to not update the internal state and keep the cell in edit mode.
+- Resolve the promise with the second argument (the original row) received to make it stop silently, not updating the internal state but exiting the edit mode.
 
-### Events [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+The next demo implements the first option.
+Instead of [validating](#validation) while typing, it validates in the server.
+If the new name is empty it will not allow to save and the cell is kept into edit mode.
+Also, it shows that `processRowUpdate` can be used to pre-process the row object that will be saved into the internal state.
+To leave the edit mode, press <kbd class="key">Escape</kbd> or enter a valid name.
 
-The editing feature leverages the event capability of the grid and the apiRef.
-The following events can be imported and used to customize the edition:
+{{"demo": "ServerSidePersistence.js", "bg": "inline", "defaultCodeOpen": false}}
```

Comment:
I missed the snakbar. Maybe having it in top-middle or bottom-middle will make it more visible than in the bottom-left which is out of the reader's attention because of the navigation

Reply:
I changed to bottom-center.